src/compiler/checker.ts

```diff
@@ -1096,9 +1096,11 @@ module ts {
                 }
 
                 // Get qualified name 
-                if (enclosingDeclaration &&
+                if ((enclosingDeclaration &&

```

Comment:
please find a way to simplify this 'if' conditional.  One option is to introduce locals above to make it clearer what you are testing for.